x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -154,7 +154,7 @@ export async function checkRunningSessions(
   try {
     await getAllSavedSearchSessions$(deps, config)
       .pipe(
-        mergeMap(async (runningSearchSessionsResponse) => {
+        concatMap(async (runningSearchSessionsResponse) => {
```

Comment:
I mentioned this in the other issue as well, but what are we hoping to accomplish by processing these serially rather than in parallel? The main issue of concern seemed to be that the update request is too large, so the only benefits I can see with serially are a decrease in CPU/memory, but I'm not sure if that will outweigh the fact that this change may also make the process take much longer.

Reply:
The whole idea of paging the sessions was to update the items in reasonable amounts and in orderly fashion.
With `mergeMap` we weren't doing that.

I think that serializing bulks of 100 search sessions makes sense, but it's going to be hard to quantify the impact.
Not 100% the same (more related to reducing the page size), but the article @Dosant linked convinced me https://eng.lifion.com/promise-allpocalypse-cfb6741298a7?gi=2c3ce0dba6ea

Reply:
@lukasolson, I think there is also a difference between grouping per 100 requests or sending all of them to ES? with `concatMap` we spread out the load.